src/node/filesystem/binary.node.ts

```diff
@@ -172,43 +178,29 @@ export class FileContainer implements IDataContainer, IBinaryFileMeta, IInitiali
   }
 
   public toStream(): NodeJS.ReadableStream {
-    if (this.stream === undefined) {
-      throw new UnexpectedError('File not initialized');
-    }
-
     if (this.streamReader === undefined) {
-      return this.stream;
+      throw new UnexpectedError('File not initialized');
     }
 
-    this.streamReader.ejectStream();
-    this.streamReader = undefined;
-
-    return this.stream;
+    return this.streamReader.toStream();
   }
 }
 
 export class StreamContainer implements IDataContainer {
   private stream: NodeJS.ReadableStream;
-  private streamReader: StreamReader | undefined;
+  private streamReader: StreamReader;
```

Comment:
question: Do we still need to explicitly hold the `stream` if we have it in `streamReader` and `toStream` is implemented through it as well?